src/fat.rs

```diff
@@ -353,47 +356,22 @@ impl<'a> OnDiskDirEntry<'a> {
     }
 
     /// If this is an LFN, get the contents so we can re-assemble the filename.
-    pub fn lfn_contents(&self) -> Option<(bool, u8, [char; 13])> {
+    pub fn lfn_contents(&self) -> Option<(bool, u8, String<U26>)> {
         if self.is_lfn() {
-            let mut buffer = [' '; 13];
             let is_start = (self.data[0] & 0x40) != 0;
             let sequence = self.data[0] & 0x1F;
-            // LFNs store UCS-2, so we can map from 16-bit char to 32-bit char without problem.
-            buffer[0] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[1..=2]))).unwrap();
-            buffer[1] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[3..=4]))).unwrap();
-            buffer[2] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[5..=6]))).unwrap();
-            buffer[3] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[7..=8]))).unwrap();
-            buffer[4] = core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[9..=10])))
-                .unwrap();
-            buffer[5] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[14..=15])))
-                    .unwrap();
-            buffer[6] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[16..=17])))
-                    .unwrap();
-            buffer[7] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[18..=19])))
-                    .unwrap();
-            buffer[8] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[20..=21])))
-                    .unwrap();
-            buffer[9] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[22..=23])))
-                    .unwrap();
-            buffer[10] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[24..=25])))
-                    .unwrap();
-            buffer[11] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[28..=29])))
-                    .unwrap();
-            buffer[12] =
-                core::char::from_u32(u32::from(LittleEndian::read_u16(&self.data[30..=31])))
-                    .unwrap();
-            Some((is_start, sequence, buffer))
+            let mut buffer = [0u8; 26];
+            buffer[..10].copy_from_slice(&self.data[1..=10]);
+            buffer[10..22].copy_from_slice(&self.data[14..=25]);
+            buffer[22..].copy_from_slice(&self.data[28..=31]);
+            let data: &[u16; 13] = unsafe { core::mem::transmute(&buffer) };
```

Comment:
Couldn't this be unaligned ?

Reply:
Compiler will ensure it's aligned by word

Reply:
Could you explain why you think this is the case ? `buffer` has an alignment of 1, while for data that is 2, why do you think that can't cause an unaligned reference ?

Reply:
look at the assembly code on x86-64 release object
```
0000000000013140 <_ZN14embedded_sdmmc3fat14OnDiskDirEntry12lfn_contents17h5e14fbe2a1624622E>:
   13140:       48 83 ec 78             sub    $0x78,%rsp
   13144:       48 89 f1                mov    %rsi,%rcx
   13147:       48 8b 76 08             mov    0x8(%rsi),%rsi
   1314b:       48 83 fe 0b             cmp    $0xb,%rsi
   1314f:       0f 86 ad 01 00 00       jbe    13302 <_ZN14embedded_sdmmc3fat14OnDiskDirEntry12lfn_contents17h5e14fbe2a1624622E+0x1c2>                                                        
   13155:       48 89 f8                mov    %rdi,%rax
   13158:       48 8b 11                mov    (%rcx),%rdx
   1315b:       8a 4a 0b                mov    0xb(%rdx),%cl
   1315e:       80 e1 0f                and    $0xf,%cl
   13161:       80 f9 0f                cmp    $0xf,%cl
   13164:       0f 85 3a 01 00 00       jne    132a4 <_ZN14embedded_sdmmc3fat14OnDiskDirEntry12lfn_contents17h5e14fbe2a1624622E+0x164>                                                        
   1316a:       44 8a 0a                mov    (%rdx),%r9b
   1316d:       0f 57 c0                xorps  %xmm0,%xmm0
   13170:       0f 11 44 24 3a          movups %xmm0,0x3a(%rsp)
   13175:       48 8b 4a 01             mov    0x1(%rdx),%rcx
   13179:       48 89 4c 24 30          mov    %rcx,0x30(%rsp)
   1317e:       0f b7 4a 09             movzwl 0x9(%rdx),%ecx
   13182:       66 89 4c 24 38          mov    %cx,0x38(%rsp)
   13187:       48 83 fe 19             cmp    $0x19,%rsi
```

It's not hard to figure out the buffer is starting from 0x30(%rsp)，which is aligned by word

Reply:
changed to definition of [u16; 13] to ensure at least 2B alignment